File: frontend/src/types/journal.ts
export type TradeRating = "excellent" | "good" | "poor";

export interface JournalEntry {
  trade_id: string;
  symbol: string;
  type: string;
  entry_price: number;
  exit_price: number;
  size: number;
  pnl: number;
  pnl_percent: number;
  duration_minutes: number;
  ai_analysis: string;
  lesson: string;
  rating: TradeRating;
  created_at: string;
}

export interface JournalResponse {
  entries: JournalEntry[];
  total: number;
}

File: backend/app/schemas/journal.py
from datetime import datetime

from pydantic import BaseModel


class JournalEntry(BaseModel):
    trade_id: str
    symbol: str
    type: str
    entry_price: float
    exit_price: float
    size: float
    pnl: float
    pnl_percent: float
    duration_minutes: int
    ai_analysis: str
    lesson: str
    rating: str          # "excellent" | "good" | "poor"
    created_at: datetime


class JournalResponse(BaseModel):
    entries: list[JournalEntry]
    total: int
